### ISSUE TITLE
Hang when awaiting RequestKernelInfo command

### ISSUE THREAD

**Issue by shyamnamboodiripad** [dotnet/interactive#2726]
### Describe the bug
```
var workingDirectoryPath = Path.Combine(Path.GetTempPath(), "interactive");
Directory.CreateDirectory(workingDirectoryPath);

// I installed the latest dotnet-interactive tool in the above directory by running the following commands
// dotnet new tool-manifest
// dotnet tool install Microsoft.dotnet-interactive --add-source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" --version 1.0.410905

var logsDirectoryPath = Path.Combine(workingDirectoryPath, "logs");
Directory.CreateDirectory(logsDirectoryPath);

var connector = new StdIoKernelConnector(
    new[] { "dotnet", $@"tool run dotnet-interactive stdio --default-kernel csharp --working-dir ""{workingDirectoryPath}"" --verbose --log-path ""{logsDirectoryPath}""" },
    KernelHost.CreateHostUri("VS"),
    new DirectoryInfo(workingDirectoryPath));

var kernel = await connector.CreateKernelAsync("Proxy");

await kernel.SendAsync(new RequestKernelInfo(kernel.KernelInfo.RemoteUri));
```

I am running into a hang inside the above `SendAsync` call for the `RequestKernelInfo` command (which means I am unable to run any subsequent commands like `SubmitCode`). The hang also happens if I omit the `destinationUri` parameter for the `RequestKernelInfo` constructor.

I was able to work around the hang by replacing the above `SendAsync` call for the `RequestKernelInfo` command with the following code that registers some commands explicitly. After this, I was able to execute `SubmitCode` commands and get back responses without any problems.
```
proxyKernel.KernelInfo.SupportedKernelCommands.Add(new(nameof(SubmitCode)));
proxyKernel.KernelInfo.SupportedKernelCommands.Add(new(nameof(RequestValue)));
proxyKernel.KernelInfo.SupportedKernelCommands.Add(new(nameof(RequestValueInfos)));
proxyKernel.KernelInfo.SupportedKernelCommands.Add(new(nameof(SendValue)));
```

**Which version of .NET Interactive are you using?** (In a notebook, run the `#!about` magic command. ): 1.0.410905

**Comment by shyamnamboodiripad** [comment]
FYI @jonsequitur I logged this issue for the problem we were discussing last week.

**Comment by colombod** [comment]
Destination uri is crucial for proxy to work. proxies should not be used directly in the absence of a host and composite, the two concur in maintaining the routing correct and setup ensuring uri are properly set in outbound commands

**Comment by jonsequitur** [comment]
We can look into fixing the API so that this isn't allowed. This might also make it necessary to look at stable kernel URIs and handling reconnection and a remote restart gesture.

**Comment by shyamnamboodiripad** [comment]
@colombod @jonsequitur Could you please clarify how I can set things up correctly when using `StdIoKernelConnector`? If you could point me to some (example / pseudo) code for how that is done that would be even better.

My assumption was that the `ProxyKernel` returned from `StdIoKernelConnector.CreateKernelAsync` can be used like any other type of kernel except for the prerequisite `RequestKernelInfo` flow (or corresponding explicit command registrations). Is the step I am missing merely the creation of a wrapping `CompositeKernel` for the returned `ProxyKernel`?

**Comment by shyamnamboodiripad** [comment]
This should all be fixed now with #2807
